### PR TITLE
@compat for f.(args...) and foo.:bar

### PR DESCRIPTION
**Pull request by stevengj**
This adds compatibility for JuliaLang/julia#15032.

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -541,6 +541,22 @@ function _compat(ex::Expr)
             callexpr.args[2] = params
             callexpr.args[3] = obj
         end
+    elseif VERSION < v"0.5.0-dev+4002" && ex.head == :. && length(ex.args) == 2 # 15032
+        if isexpr(ex.args[2], :quote) && typeof(ex.args[2].args[1]) == QuoteNode
```
Comment (timholy) [inline src/Compat.jl]: `isa(x, QuoteNode)`

**Comment by timholy** [comment]
LGTM. The `isa` is basically stylistic, so feel free to go ahead and merge if you prefer.

**Comment by stevengj** [comment]
Changed to `isa`.  Merge when green.

**Comment by timholy** [comment]
Thanks!

**Pull request merged**